### PR TITLE
feat: allow adding custom headers with nuxt config

### LINKED ISSUE #146: Allow headers to be set in nuxt.config.js
### What problem does this feature solve?

I am interested in being able to set default global headers without extending axios. Ideally, I'd like to do:

```
axios: {
  header: { 'Cockpit-Token': '2cbe585716f6XXXXXXXXXXX' }
},
```

and it'd default to `common` as well as merge the defaults set within' `plugin.template.js` as well as be able to specify the specific header such as:

```
axios: {
  header: { get: { 'Cockpit-Token': '2cbe585716f6XXXXXXXXXXX' } }
},
```

This would be greatly appreciated as not every headless CMS, for example, uses the `Authorization` header, and having to specify it for every request can get a little cumbersome. 

If this is already possible, an example would be much appreciated as I sat here for a few hours trying just about every combination I could possibly think of. 

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/axios-module/issues/c136">#c136</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
Resolve #146

### CODE REVIEW

File: docs/options.md [hunk 1]
```diff
@@ -123,3 +123,23 @@ Also helps making consistent requests in both SSR and Client Side code.
...
+    delete: {},
+    get: {},
+    head: {},
+    post: {},
+    put: {},
+    patch: {}
+}
+```
+
+You can customize the headers. Your setting will be merged with the default setting.
```
Comment (pi0) [inline docs/options.md]: You can customize (default) headers

File: lib/module.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const path = require('path')
 const consola = require('consola')
+const merge = require('deepmerge')
```
Comment (pi0) [inline lib/module.js]: Can we use defu instead?

File: lib/plugin.js [hunk 3]
```diff
@@ -156,17 +156,7 @@ export default (ctx, inject) => {
...
-      'Accept': 'application/json, text/plain, */*'
-    },
-    delete: {},
-    get: {},
-    head: {},
-    post: {},
-    put: {},
-    patch: {}
-  }
+  const headers = <%= serialize(options.headers) %>
```
Comment (pi0) [inline lib/plugin.js]: I would suggest using JSON.stringify with proper format

**Comment by pi0** [comment]
Thanks mate :)

**Comment by A5rocks** [comment]
Hello! Just writing this mainly as a spot for people to find from search. AFAICT you cannot remove headers from the default `proxyHeadersIgnore` due to this change -- `defu` concatenates arrays together. This means if you provide a list without a certain header which is in the default list... your change is futile.

A question for PR author / @pi0: was this change on purpose? The only way to remove `CF-Connecting-IP` from the list seems to be to use `patch-package`. Needless to say, I don't like this "fix".

**Pull request merged**